pkg/controller/tas/resource_flavor_test.go

```diff
@@ -0,0 +1,169 @@
+				ConditionHeartbeat(corev1.NodeReady, later).
+				ConditionHeartbeat(corev1.NodeMemoryPressure, later).
+				Obj(),
+			wantChanged: false,
+		},
+		"Annotation changed": {
+			oldNode:     baseNode.Clone().Obj(),
+			newNode:     baseNode.Clone().Annotation("new-annotation", "new-value").Obj(),
+			wantChanged: true,
+		},
+		"Label changed": {
+			oldNode:     baseNode.Clone().Obj(),
+			newNode:     baseNode.Clone().Label("new-label", "new-value").Obj(),
+			wantChanged: true,
+		},
+		"Node Ready status changed": {
+			oldNode: baseNode.Clone().Obj(),
+			newNode: baseNode.Clone().
+				ConditionStatus(corev1.NodeReady, corev1.ConditionFalse, later).
+				Obj(),
+			wantChanged: true,
+		},
+		"Allocatable resources changed": {
+			oldNode: baseNode.Clone().Obj(),
+			newNode: baseNode.Clone().StatusAllocatable(corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("16"),
+				corev1.ResourceMemory: resource.MustParse("32Gi"),
+			}).Obj(),
+			wantChanged: true,
+		},
+		"Taints changed": {
+			oldNode: baseNode.Clone().Obj(),
+			newNode: baseNode.Clone().Taints(corev1.Taint{
+				Key:    "new-taint",
+				Value:  "new-value",
+				Effect: corev1.TaintEffectNoExecute,
+			}).Obj(),
+			wantChanged: true,
+		},
+		"Taints with TimeAdded": {
+			oldNode: baseNode.Clone().Taints(corev1.Taint{
+				Key:       "test",
+				Value:     "value",
+				Effect:    corev1.TaintEffectNoExecute,
+				TimeAdded: &now,
+			}).Obj(),
+			newNode: baseNode.Clone().Taints(corev1.Taint{
+				Key:       "test",
+				Value:     "value",
+				Effect:    corev1.TaintEffectNoExecute,
+				TimeAdded: &later,
+			}).Obj(),
+			wantChanged: true,
+		},
+		"Unschedulable changed": {
+			oldNode:     baseNode.Clone().Obj(),
+			newNode:     baseNode.Clone().Unschedulable().Obj(),
+			wantChanged: true,
+		},
+		"Update Multiple properties": {
+			oldNode: baseNode.Clone().Obj(),
+			newNode: baseNode.Clone().
+				ConditionHeartbeat(corev1.NodeReady, later).
+				ConditionHeartbeat(corev1.NodeMemoryPressure, later).
+				Annotation("another-annotation", "another-value").
+				ResourceVersion("12345").
+				Obj(),
+			wantChanged: true,
+		},
+		"New condition type added": {
+			oldNode: baseNode.Clone().Obj(),
+			newNode: baseNode.Clone().StatusConditions(corev1.NodeCondition{
+				Type:               corev1.NodeDiskPressure,
+				Status:             corev1.ConditionTrue,
+				LastHeartbeatTime:  now,
+				LastTransitionTime: now,
+			}).Obj(),
+			wantChanged: true,
+		},
+		"Condition removed": {
+			oldNode: baseNode.Clone().StatusConditions(corev1.NodeCondition{
+				Type:               corev1.NodeDiskPressure,
+				Status:             corev1.ConditionFalse,
+				LastHeartbeatTime:  now,
+				LastTransitionTime: now,
+			}).Obj(),
+			newNode:     baseNode.Clone().Obj(),
+			wantChanged: true,
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			got := checkNodeSchedulingPropertiesChanged(tc.oldNode, tc.newNode)
```

Comment:
Oh, I got it. Ideally, we want to call `Update` function here.
But, we need to implement the fake workqueue RateLimiter for that.

So, I'm ok with testing for `checkNodeSchedulingPropertiesChanged` since I think that implementing fake workqueue is extra work.

Reply:
If you can create an issue, I would appreciate that.